chore: enable pre-releases for subscriptions

Roming22 opened the pull request:
This script should be useful to developers and QE. It might be useful to the occasional customer that would want to validate an upcoming fix in an ephemeral environment.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Introduced a script to automate the setup of pre-release subscriptions for multiple products in OpenShift, including Red Hat Developer Hub, GitOps, Pipelines, and Trusted Artifact Signer.
  * Added support for command-line options to select products, enable debug mode, and display help.
  * Enhanced error handling and progress messaging during the configuration process.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

xinredhat commented on hack/prerelease.sh:
```diff
@@ -0,0 +1,136 @@
+                PRODUCT_LIST+=( "$2" )
+                ;;
+            *)
+                echo "[ERROR] Unknown product: $1"
```
The variable should be `$2`

Roming22 replied:
Done

xinredhat commented on hack/prerelease.sh:
```diff
@@ -0,0 +1,136 @@
+                ;;
+            *)
+                echo "[ERROR] Unknown product: $1"
+                usage
```
when the product is invalid, it should exist the script. So here it's missing  `exit 1`

Roming22 replied:
Done

xinredhat commented on hack/prerelease.sh:
```diff
@@ -0,0 +1,136 @@
+    ${0##*/} [options]
+
+Optional arguments:
+    -p, --product $PRODUCT
```
I think it shouldn't use a variable `$PRODUCT`,  it will fail with error `PRODUCT: unbound variable`

Roming22 replied:
Done

xinredhat commented on hack/prerelease.sh:
```diff
@@ -0,0 +1,136 @@
+    PRODUCT_LIST=()
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+        -p|--product)
```
There's no check to verify that $2 exists when -p/--product is specified. This could cause an error if someone runs -p without a product name. 
```suggestion
        -p|--product)
            if [[ -z "${2:-}" ]]; then
                echo "[ERROR] Product name required for $1 option"
                usage
                exit 1
            fi
```

Roming22 replied:
Done

Roming22 commented:
/retest

xinredhat commented:
/retest

Roming22 commented:
/retest

dperaza4dustbit commented on hack/pre-release.sh:
```diff
@@ -0,0 +1,221 @@
+    DOCKERCONFIGJSON=$(
+        echo "$DOCKERCONFIGJSON" \
+        | jq -c '.auths."registry.stage.redhat.io" = {
+            "auth": "MTEwMDkxMDN8cmhhZHM6ZXlKaGJHY2lPaUpTVXpVeE1pSjkuZXlKemRXSWlPaUk0T0RsaE1XWTJPR0ZsWVdVMFptTTNZVE13WWpBNVlXUmpOR1JsWVRneVlpSjkuSXZrVDZqQ001ZU9wUnJQeDVzOVk4NFlaY1NfcWpDX0RpZ0s1bmwtd3NCQ0Y4cGFVdG13b25RbE5GTm9sYXBQTGZUeFFUaWxUUDluYllIRGV4TGxGWnZUWGNRWmg4YllqMmFGOUhZaHR2bFpCbmxpS0tyakNVN24tUE5rb09DY296QmdfN21fZFU4Z1lIUGt4Q3pISC1obG53Y2dneklwMG5jV0xQTDhFTkM4R0lJV2pQZ0t4bm1vTDQ2TUNUUD ...
```
Is this supposed to be a secret looks like a jwt token?

Roming22 replied:
I'm surprised by the format as well, but that's what was shared by the TAS team, and it was working fine in my tests.

dperaza4dustbit reviewed (commented):
One question

Roming22 committed:
chore: enable pre-releases for subscriptions

This script should be useful to developers and QE.
It might be useful to the occasional customer that would want to
validate an upcoming fix in an ephemeral environment.

lcarva commented:
Is this still relevant?

Roming22 commented:
@lcarva While the specific versions might not be, I believe the process is.

jkopriva commented:
/lgtm
